libstuff/libstuff.h

```diff
@@ -230,14 +230,18 @@ void SSyslogSocketDirect(int priority, const char* format, ...);
 // Atomic pointer to the syslog function that we'll actually use. Easy to change to `syslog` or `SSyslogSocketDirect`.
 extern atomic<void (*)(int priority, const char *format, ...)> SSyslogFunc;
 
+//template <typename T>
+//string addLogParams(const string& message, const map<string, T> params);
+
+string addLogParams(const string& message, const map<string, string>& params);
+string addLogParams(const string& message);
+
 // **NOTE: rsyslog default max line size is 8k bytes. We split on 7k byte boundaries in order to fit the syslog line prefix and the expanded \r\n to #015#012
 #define SWHEREAMI SThreadLogPrefix + "(" + basename((char*)__FILE__) + ":" + SToStr(__LINE__) + ") " + __FUNCTION__ + " [" + SThreadLogName + "] "
-#define SSYSLOG(_PRI_, _MSG_)                                                   \
+#define SSYSLOG(_PRI_, _MSG_, ...)                                              \
     do {                                                                        \
         if (_g_SLogMask & (1 << (_PRI_))) {                                     \
-            ostringstream __out;                                                \
-            __out << _MSG_ << endl;                                             \
-            const string s = __out.str();                                       \
+            const string s = addLogParams(_MSG_, ##__VA_ARGS__);                \
```

Comment:
This is varaidic to allow it to be empty?

Reply:
Yes